.github/workflows/test.yml

```
@@ -6,10 +6,36 @@ on:
     types: [opened, synchronize]
     branches-ignore: [staging, production]
 
+env:
+  # Number of parallel jobs for jest tests
+  CHUNKS: 3
 jobs:
+  config:
+    runs-on: ubuntu-latest
+    name: Define matrix parameters
+    outputs:
+      MATRIX: ${{ steps.set-matrix.outputs.MATRIX }}
+    steps:
+      - name: Set Matrix
+        id: set-matrix
+        uses: actions/github-script@v6
+        with:
+          # Generate matrix array i.e. [0, 1, 2, ...., CHUNKS] for test job
+          script: |
+            core.setOutput('MATRIX', Array.from({ length: Number(process.env.CHUNKS) + 1 }, (v, i) => i));
+
   test:
+    needs: config
     if: ${{ github.actor != 'OSBotify' || github.event_name == 'workflow_call' }}
     runs-on: ubuntu-latest
+    name: test (job ${{ matrix.chunk }})
```

```suggestion
    name: test (job ${{ matrix.chunk + 1 }})
```
Make sure to convert it to number if `matrix.chunk` is string here.

---

@parasharrajat I pushed this change
```
name: test (job ${{ fromJSON(matrix.chunk) + 1 }})
```

Do you want ant other change? If I missed any requested changes, do let me know, I will do that